src/main/java/com/amazonaws/encryptionsdk/internal/FrameDecryptionHandler.java

```diff
@@ -133,6 +133,11 @@ public ProcessingSummary processBytes(final byte[] in, final int off, final int
                 int protectedContentLen = -1;
                 if (currentFrameHeaders_.isFinalFrame()) {
                     protectedContentLen = currentFrameHeaders_.getFrameContentLength();
+
+                    // The final frame should not be able to exceed the frameLength
+                    if(frameSize_ > 0 && protectedContentLen > frameSize_) {
```

Comment:
frame size can be 0 and the final frame content length can be strictly equal to the frame size.

Reply:
I'm only throwing the exception if the final frame size is strictly greater than the frame size. If the frame size is 0, then I can't compare the last frame size so I don't throw the exception in that case.

Reply:
According to https://github.com/aws/aws-encryption-sdk-java/blob/163fd8847760a0f3e3af473530f8f7ccf03ae910/src/main/java/com/amazonaws/encryptionsdk/internal/FrameDecryptionHandler.java#L122 frameSize_ _can_ be 0, but from https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/message-format.html#header-frame-length it is not the intention of the frameLength to be 0 in the header.

Reply:
@SalusaSecondus do you know if there a reason we allow the frame size in the header to be 0?

Reply:
This code *should* never execute if the frame length is 0.

encryption:
https://github.com/aws/aws-encryption-sdk-java/blob/163fd8847760a0f3e3af473530f8f7ccf03ae910/src/main/java/com/amazonaws/encryptionsdk/internal/EncryptionHandler.java#L120-L146

decryption:
https://github.com/aws/aws-encryption-sdk-java/blob/163fd8847760a0f3e3af473530f8f7ccf03ae910/src/main/java/com/amazonaws/encryptionsdk/internal/DecryptionHandler.java#L497-L505

The comment on the quoted lines makes me think that this was something that was added in a long time ago, possibly for a very early version of the message format because the frame size is not included in the frame header.

https://github.com/aws/aws-encryption-sdk-java/blob/163fd8847760a0f3e3af473530f8f7ccf03ae910/src/main/java/com/amazonaws/encryptionsdk/internal/FrameDecryptionHandler.java#L123-L124

Reply:
Is it possible for the ciphertext to be corrupted/manipulated in some way that leaves the frame size as 0, or would that be caught upstream and cause decryption to fail?

Reply:
I've added validation to ensure the frame size is not zero if the data is framed

Reply:
From discussing with @SalusaSecondus, frame size = 0 in the header was intended to indicate a variable length encoding. This feature was not fully implemented and is not documented, so it should be removed, but there are more remnants of it scattered in other parts of the code, so I've opened #167 to track that. For this PR I will only validate the final frame length if the header frame length is > 0